types/timers.d.ts

```diff
@@ -0,0 +1,71 @@
+declare module "timers" {
+  export import setTimeout = globalThis.setTimeout;
+  export import clearTimeout = globalThis.clearTimeout;
+  export import setInterval = globalThis.setInterval;
+  export import clearInterval = globalThis.clearInterval;
+  export import setImmediate = globalThis.setImmediate;
+
+  global {
+    /**
+     * This object is created internally and is returned from `setTimeout()` and `setInterval()`. It can be passed to either `clearTimeout()` or `clearInterval()` in order to cancel the
+     * scheduled actions.
+     */
+    class Timeout {}
+
+    /**
+     * Schedules execution of a one-time `callback` after `delay` milliseconds.
+     *
+     * The `callback` will likely not be invoked in precisely `delay` milliseconds.
+     * Caido makes no guarantees about the exact timing of when callbacks will fire,
+     * nor of their ordering. The callback will be called as close as possible to the
+     * time specified.
+     *
+     * When `delay` is less than `4`, the `delay` will be set to `4`.
```

Comment:
This is not true in the current code, but I feel like it should? @richarddavison thoughts

Reply:
Not really. It's a bit more complex than that. If one timer has been executed and the next one is scheduled to be executed in 1-3ms it will be rounded up to 4ms. If it's scheduled time already has passed it will be executed immediately. We can write something like "the precision of timers are 4ms". This is ofc something we in the future can override or set via an env if needed to. But it feels pretty OK for now

Reply:
Will do sorry been very busy